src/components/List/HierarchyList/HierarchyList.jsx

```diff
@@ -87,25 +87,98 @@ export const searchItem = (item, searchTerm) => {
  * @param {String} value what to search for
  * @returns {Array<Object>}
  */
-export const searchNestedItems = (items, value) => {
+export const searchForNestedItemValues = (items, value) => {
   const filteredItems = [];
   cloneDeep(items).forEach(item => {
     // if the item has children, recurse and search children
     if (item.children) {
       // eslint-disable-next-line
-      item.children = searchNestedItems(item.children, value);
+      item.children = searchForNestedItemValues(item.children, value);
       // if it's children did, we still need the item
       if (item.children.length > 0) {
         filteredItems.push(item);
       }
     } // if the item matches, add it to the filterItems array
-    else if (searchItem(item, value)) {
+    else if (searchForItemValue(item, value)) {
       filteredItems.push(item);
     }
   });
   return filteredItems;
 };
 
+/**
+ * Searches an item for a specific id
+ * @param {Object} item to be searched
+ * @param {string} searchId
+ * @returns {Boolean} found or not
+ */
+export const searchForItemId = (item, searchId) => {
+  if (
+    // Check if the value or secondary value has a match
+    item.id.toLowerCase().search(searchId.toLowerCase()) !== -1
```

Comment:
Can't this function and searchForItemValue be refactored into one to take a list of content keys to search for so the same codebase can be shared?

Reply:
Refactored and made a general util search function